src/NHibernate/Type/AbstractDateTimeType.cs

```diff
@@ -0,0 +1,160 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Data.Common;
+using System.Globalization;
+using NHibernate.Engine;
+using NHibernate.SqlTypes;
+
+namespace NHibernate.Type
+{
+	/// <summary>
+	/// Base class for date time types.
+	/// </summary>
+	[Serializable]
+	public abstract partial class AbstractDateTimeType : PrimitiveType, IIdentifierType, ILiteralType, IVersionType
```

Comment:
New base type for types yielding .Net `DateTime`, including `DateType`.

It includes the logic of the removed `AbstractDateTimeSpecificKindType`, which was applied only to type with kind `DateTimeKind.Local` or `DateTimeKind.Utc`, not to other types (non prefixed with `Local` or `Utc`). Thus here, we take into `Kind` only if it not `Unspecified`, its default value in this base class.